Refactors zookeeper dependencies and Kafka span receiver test

codefromthecrypt opened the pull request:
Zookeeper dependencies conflicted in most contexts. I set the minimum
version to coincide with the minimum used at Twitter internally, which
matches what Kakfa and Curator use.

Next, this ports the existing Kafka span receiver test to use a JUnit
rule, as opposed to manually orchestrating kafka and zookeeper. This
requires at least JUnit 4.11 for scala interop.

codefromthecrypt commented on project/Project.scala:
```diff
@@ -379,9 +383,10 @@ object Zipkin extends Build {
         "org.apache.commons" % "commons-io" % "1.3.2",
-        scroogeDep("serializer")
+        scroogeDep("serializer"),
+        "com.github.charithe" % "kafka-junit" % "1.3" % "test"
```
@abesto everything above the kafka-junit line should be done to the build regardless of my change

abesto replied:
Gotcha. Ping me once this is merged and I'll take care of the conflict accordingly.

codefromthecrypt commented:
rebased over latest decoupled build

codefromthecrypt commented:
@jamescway PTAL

eirslett commented:
You can use `@Rule` in Scala if you use JUnitSuite:
http://doc.scalatest.org/1.7/org/scalatest/junit/JUnitSuite.html

Then you won't need the painful `$.MODULE$` stuff that comes from bad Java-Scala-interop... (I'm working on making Span more Java-friendly!)

codefromthecrypt commented:
Thx will change it!

Commit:
Refactors zookeeper dependencies and Kafka span receiver test

Zookeeper dependencies conflicted in most contexts. I set the minimum
version to coincide with the minimum used at Twitter internally, which
matches what Kakfa and Curator use.

Next, this ports the existing Kafka span receiver test to use a JUnit
rule, as opposed to manually orchestrating kafka and zookeeper. This
requires at least JUnit 4.11 for scala interop.

codefromthecrypt commented:
@eirslett ok.. was a little tricky, but yeah better than interop on the Span type for sure. PTAL!

codefromthecrypt commented on zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:
```diff
@@ -1,47 +1,29 @@
-import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
-import com.twitter.server.TwitterServer
-import com.twitter.zipkin.zookeeper.ZooKeeperClientFactory
+  @ClassRule def kafkaRuleDef = kafkaRule
```
this dance is to ensure the rule is a singleton.

rtyler replied:
why not put that comment into the code itself for posterity :)

codefromthecrypt replied:
roger that!

codefromthecrypt replied:
done

rtyler commented:
an timid :+1: 

I think @eirslett should sign off and merge this if he's good with it

jamescway commented on zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:
```diff
@@ -1,47 +1,31 @@
-import java.io._
+object KafkaProcessorSpecSimple {
+  // Singleton as the test needs to read the actual port in use
+  val kafkaRule = new KafkaJunitRule()
```
Pretty cool, [kafka-unit](https://github.com/chbatey/kafka-unit) allows you to delete all that test code

codefromthecrypt replied:
thank @charithe!

jamescway commented:
:+1: :+1:  (u get 2)  super awesome!

codefromthecrypt commented:
@eirslett forgive me for merging :) I think I addressed your concern and happy to revise further in a latter change as necessary!

eirslett commented:
Great work! I love that you could delete all that code!
`kafkaRule` could maybe be a `lazy val`, but whatever.
Cheers! :beers:
